code/game/alternate_appearance.dm

```diff
@@ -113,6 +113,19 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
 		return TRUE
 	return FALSE
 
+/datum/atom_hud/alternate_appearance/basic/AI
+
+/datum/atom_hud/alternate_appearance/basic/AI/New()
+	..()
+	for(var/mob/living/silicon/ai/AI in GLOB.ai_list)
+		if(mobShouldSee(AI))
```

Comment:
make the loop typeless

Reply:
Also since your `mobShouldSee` just checks if the mob is an AI, the call to it is unecessary

Reply:
Right, I only copied the format of the older procs.